src/util/mod.rs

```diff
@@ -38,3 +38,16 @@ pub fn collapse<A>(r: Result<A, A>) -> A {
         Err(e) => e,
     }
 }
+
+pub trait ErrInto<D> {
+    fn err_into(self) -> D;
+}
+
+impl<SrcErr, DestErr, DestRes> ErrInto<Result<DestRes, DestErr>> for Result<DestRes, SrcErr>
+where
+    DestErr: From<SrcErr>,
+{
+    fn err_into(self) -> Result<DestRes, DestErr> {
+        self.map_err(|e| e.into())
```

Comment:
there are two standard traits for type conversion, `Into` and `From`. This mechanism depends on the error type already implementing the conversion trait. The `?`-operator will already apply conversions, so I don't think you need this trait or this function at all.

Reply:
An explicit conversion is needed.